homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,274 @@
+    STATE_STREAMING,
+    SUPPORT_ON_OFF,
+    SUPPORT_STREAM,
+)
+from homeassistant.components.mjpeg.camera import (
+    CONF_MJPEG_URL,
+    CONF_STILL_IMAGE_URL,
+    MjpegCamera,
+    filter_urllib3_logging,
+)
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
+from homeassistant.helpers import entity_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    ATTRIBUTION,
+    CAMERA_SCAN_INTERVAL_SECS,
+    CONNECTION,
+    DOMAIN as AGENT_DOMAIN,
+    SERVICE_UPDATE,
+)
+from .helpers import service_signal
+
+SCAN_INTERVAL = timedelta(seconds=CAMERA_SCAN_INTERVAL_SECS)
+
+_LOGGER = logging.getLogger(__name__)
+
+_DEV_EN_ALT = "enable_alerts"
+_DEV_DS_ALT = "disable_alerts"
+_DEV_EN_REC = "start_recording"
+_DEV_DS_REC = "stop_recording"
+_DEV_SNAP = "snapshot"
+
+CAMERA_SERVICES = {
+    _DEV_EN_ALT: "async_enable_alerts",
+    _DEV_DS_ALT: "async_disable_alerts",
+    _DEV_EN_REC: "async_start_recording",
+    _DEV_DS_REC: "async_stop_recording",
+    _DEV_SNAP: "async_snapshot",
+}
+
+
+async def async_setup_entry(
+    hass, config_entry, async_add_entities, discovery_info=None
+):
+    """Set up the Agent cameras."""
+    filter_urllib3_logging()
+    cameras = []
+
+    server = hass.data[AGENT_DOMAIN][config_entry.entry_id][CONNECTION]
+    server.cameras = []
+    if not server.deviceList:
+        _LOGGER.warning("Could not fetch cameras from Agent server")
+        return
+
+    for device in server.deviceList:
+        if device.typeID == 2:
+            camera = AgentCamera(device)
+            cameras.append(camera)
+            server.cameras.append(camera)
+
+    async_add_entities(cameras)
+
+    platform = entity_platform.current_platform.get()
+    for service, method in CAMERA_SERVICES.items():
+        platform.async_register_entity_service(service, {}, method)
+
+    return True
+
+
+class AgentCamera(MjpegCamera):
+    """Representation of an Agent Device Stream."""
+
+    def __init__(self, device, enabled_default: bool = True):
+        """Initialize as a subclass of MjpegCamera."""
+        self._servername = device.client.name
+        self.server_url = device.client._server_url
+
+        device_info = {
+            CONF_NAME: device.name,
+            CONF_MJPEG_URL: f"{self.server_url}{device.mjpeg_image_url}&size=640x480",
+            CONF_STILL_IMAGE_URL: f"{self.server_url}{device.still_image_url}&size=640x480",
+        }
+        self.device = device
+        self._removed = False
+        self._unsub_dispatcher = []
+        self._name = f"{self._servername} {device.name}"
+        self._stream_url = f"{self.server_url}{device.mp4_url}"
+        self._unique_id = f"{device._client.unique}_{device.typeID}_{device.id}"
+        self._enabled_default = enabled_default
+        super().__init__(device_info)
+
+    @property
+    def should_poll(self) -> bool:
+        """Update the state periodically."""
+        return True
+
```

Comment:
```suggestion
```

Should Poll is the default

Reply:
What if the default changes? That will break the integration.

Reply:
The default is not likely to change as that would break most of the integrations.

Reply:
Are there guidelines for these reviews i'm not aware of? I don't see the problem with setting should_poll to True - it's completely trivial, it's just there in case the default value changes. It might be "not likely to change" but that isn't really good enough is it?

Reply:
The default for camera entities is to not poll:
https://github.com/home-assistant/core/blob/14190bd497609a09e8e7134ab941a5a11e122335/homeassistant/components/camera/__init__.py#L326-L329

Reply:
This will be why it's not polling now. I'll add it back in. This review process is agony.